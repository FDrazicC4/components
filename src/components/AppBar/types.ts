import React from 'react';
import { Colors } from '../types';


export interface Props {


    /**
     * The position in which the element is anchored from by default, responsiveness manipulate by css styling
     */
    anchorFrom?: 'top' | 'right' | 'bottom' | 'left',


    /**
     * Custom classname added to root
     */
    className?: string,


    /**
     * Any valid JSX
     */
    children?: React.ReactNode,


    /**
     * Background color of the element
     */
    color?: Colors,


}


export interface AppBarComponent extends React.ForwardRefExoticComponent<any> {
    Item?: any,
}
