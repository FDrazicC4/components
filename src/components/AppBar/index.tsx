import React from 'react';
import clsx from 'clsx';
import MuiAppBar from '@material-ui/core/AppBar';
import MuiToolbar from '@material-ui/core/Toolbar';
import { Props } from './types';


export const AppBar = React.forwardRef(({
    anchorFrom = 'left',
    children,
    className,
    color = 'dark',
    ...otherProps
}: Props): React.ReactElement => {


    const classes = {
        root: `brew-AppBar brew-AppBar--${color} brew-AppBar--${anchorFrom} ${className}`,
    };


    return (
        <MuiAppBar
            classes={classes}
            {...otherProps}
        >
            <MuiToolbar className="brew-AppBar__content">
                {children}
            </MuiToolbar>
        </MuiAppBar>
    );


});


AppBar.displayName = 'AppBar';


export default AppBar;
