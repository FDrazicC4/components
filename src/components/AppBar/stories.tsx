import React from 'react';
import { select } from '@storybook/addon-knobs';
import Flag from '@components/Flag';
import Icon from '@components/Icon';
import AppBar from '@components/AppBar';
import { Colors as ColorOptions } from '../types';


export default {
    component: AppBar,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|Data Display/AppBar',
};


export const Sandbox = (): React.ReactElement => {


    const color: ColorOptions = select(
        'color',
        ['primary', 'secondary', 'tertiary', 'light', 'dark'],
        'dark'
    );

    const anchorFrom = select(
        'anchorFrom',
        ['top', 'right', 'bottom', 'left'],
        'left',
    );


    return (
        <AppBar
            anchorFrom={anchorFrom}
            className="custom-appbar-class"
            color={color}
        >
            <Icon color="primary" size="large">add_circle</Icon>
            <Icon color="primary" size="large">contacts</Icon>
            <Icon color="primary" size="large">business</Icon>
            <Icon color="primary" size="large">add</Icon>
            <Icon color="primary" size="large">more_vert</Icon>
        </AppBar>
    );


};
