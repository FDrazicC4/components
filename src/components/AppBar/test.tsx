import React from 'react';
import renderer from 'react-test-renderer';
import AppBar from '@components/AppBar';


it('renders correctly', () => {
    const tree = renderer
        .create(<AppBar>Foo</AppBar>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
