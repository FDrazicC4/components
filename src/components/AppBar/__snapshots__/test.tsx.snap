// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root brew-AppBar brew-AppBar--left MuiAppBar-positionFixed MuiAppBar-colorInherit mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular brew-AppBar__content MuiToolbar-gutters"
  >
    Foo
  </div>
</header>
`;
